src/backend/replication/walreceiver.c

```diff
@@ -1091,13 +1115,18 @@ XLogWalRcvClose(XLogRecPtr recptr)
 						xlogfname)));
 
 	/*
+	 * Whenever archive_mode is always or streaming of archival status from primary is enabled,
+	 * we create a .ready file because standby can get promoted before primary has archived.
+	 * It will be replaced by .done later when we get an acknowledgement from primary that
+	 * this has already been archived.
+	 *
 	 * Create .done file forcibly to prevent the streamed segment from being
 	 * archived later.
```

Comment:
Best to synchronize the comment as well.

```suggestion
	 * Create .done file forcibly to prevent the streamed segment from being
	 * being archived later unless ARCHIVE_MODE_ALWAYS is set.
	 *
	 * GPDB: The wal_sender_archiving_status_interval GUC will make it
	 * so that .ready files are created instead of .done files if
	 * nonzero and an archive command is set. The .ready files will be
	 * transitioned to .done after receiving WAL archiving status
	 * update messages from the primary segment.
	 */
```

Reply:
Done